src/contexts/APIRouterContext/Provider.tsx

```diff
@@ -186,6 +187,44 @@ const APIRouterProvider: React.FC = ({ children }) => {
     [_handlePlaceOrder, accountRs, publicKey]
   );
 
+  const _handleSetAccountInfo = useCallback(
+    async (tx: ISetAccountInfo, secretPhrase: string) => {
+      tx.secretPhrase = secretPhrase;
+      let result;
+
+      try {
+        result = await setAccountInfo(tx);
+      } catch (e) {
+        console.error("error while setting account info:", e);
+        return;
+      }
+
+      if (!result) {
+        enqueueSnackbar(messageText.userInfo.failure, { variant: "error" });
+        return;
+      }
+
+      enqueueSnackbar(messageText.userInfo.success, { variant: "success" });
+    },
+    [enqueueSnackbar]
+  );
+
+  const handleSetAccountInfo = useCallback(
+    async (accountName: string, accountDescription: string): Promise<true | undefined> => {
+      const tx: ISetAccountInfo = {
+        name: accountName,
+        description: accountDescription,
+        secretPhrase: "",
+      };
+
+      afterSecretCB.current = _handleSetAccountInfo.bind(null, tx);
```

Comment:
we may need one more ref that is set here to help gate the state of the cb ref. so that we can't get into a situation where the user takes forever to complete the prompts and then some underlying timer or block update changes the callback to point to another action instead. it's essentially a race condition check/enforcement that until the callback has run, no new callback can be set to it.

we may end up putting a timeout on that logic too, depending on needs, but not for now. just the boolean check.

Reply:
The dialog takes over the screen so I don't think I understand how it could be a problem.

Reply:
it's a non-issue for now after further review